mlir/lib/Dialect/Mesh/Transforms/Spmdization.cpp

```diff
@@ -796,8 +790,12 @@ spmdizeFuncOp(FunctionOpInterface op, IRMapping &spmdizationMap,
   // Snapshot the original blocks to not mess up the iteration when adding new
   // blocks.
   SmallVector<Block *> originalBlocks;
-  llvm::transform(op.getBlocks(), std::back_inserter(originalBlocks),
-                  [](Block &b) { return &b; });
+  for (Block &b : op.getBlocks()) {
```

Comment:
So, if you have two blocks with shard ops and one in the middle without (but still dominating the last one), you don't SPMDize the middle block?

Reply:
Without `ShardOp`s there is nothing to spmdize. The spmdize machinery however expects fully annotated shardings. 
Non-annotated blocks are expected to operate scalar-like.